feat(StageChannel): add createStageInstance method & use better naming convention

iShibi opened the pull request:
**Please describe the changes this PR makes and why it should be merged:**

This PR adds a new `createStageInstance` method on `StageChannel` that lets the user create a stage instance associated to it. It also refactors the options typedef to keep it consistent with rest of the lib:

1) `CreateStageInstanceOptions` is now `StageInstanceCreateOptions`
2) The `channel` field is a seperate param now instead of being inside the options object

⚠️ As pointed out by @SpaceEEC, the term **"instance"** makes it sound like it is an instance of a Stage Channel which it isn't. This PR removes this ambiguity by using the complete term i.e. **"Stage Instance"** in all places. 

**Status and versioning classification:**
- Code changes have been tested against the Discord API
- I know how to update typings and have done so
- This PR changes the library's interface
- This PR includes breaking changes
<!--
Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- This PR **only** includes non-code changes, like changes to documentation, README, etc.
-->

kyranet reviewed (commented):
I don't agree with the breaking change as the object still has a required property (`topic`).

If it's because `StageChannel#createStageInstance` (probably should be `createInstance`), you can do `create({ ...options, channel: this.id })`.

iShibi commented:
It will be the same situation like `MessageOptions` again, where `messageReference` was getting set to `this.id` and hence `ReplyMessageOptions` was created. Unlike there, where we couldn't make `messageReference` a seperate param, it is much more intuitive to have `channel` as a first param to `stageInstances#create` here.

`createInstance` sounds good tho, will rename it.

SpaceEEC commented on src/managers/StageInstanceManager.js:
```diff
@@ -28,31 +28,30 @@ class StageInstanceManager extends BaseManager {
   /**
    * Creates a new stage instance.
-   * @param {CreateStageInstanceOptions} options The options to create the stage instance
+   * @param {StageChannel|Snowflake} channel The stage channel whose instance is to be created
```
See my comment below:
```suggestion
   * @param {StageChannel|Snowflake} channel The stage channel to associate the created instance to
```

iShibi commented on src/structures/StageChannel.js:
```diff
@@ -28,6 +28,15 @@ class StageChannel extends BaseGuildVoiceChannel {
   }
 
+  /**
+   * Creates an instance of this stage channel.
```
Prior to this, the word `instance` was "supposed" to be short for `stage instance` but now that you pointed it out, using the short term does allow opportunities for making some wrong assumptions here. **"Stage Instance of a Stage Channel"** is much unambiguous than **"Instance of a Stage Channel"**. Can you add "stage" before the "instance" in these two suggestions so I can commit them now. I'll make necessary changes in other places. Thanks for noticing this 😸

iShibi commented:
Refactored some stuff to reflect the new naming convention for things related to stage instances. Added more information covering this change in the description above and changed the title.

*"instance of stage channel" What was I thinking pepehands*

vladfrangu commented on src/managers/StageInstanceManager.js:
```diff
@@ -28,31 +28,30 @@ class StageInstanceManager extends BaseManager {
-  async create(options) {
-    if (typeof options !== 'object') throw new TypeError('INVALID_TYPE', 'options', 'object', true);
-    let { channel, topic, privacyLevel } = options;
+  async create(channel, options) {
```
Same as kyra, I do not understand why channel is now outside of the options object, and personally I'd prefer it be a full object. Thoughts @discordjs/the-big-3?

kyranet replied:
The unified approach is consistent with [`RoleManager`](https://discord.js.org/#/docs/main/master/class/RoleManager?scrollTo=create), [`ThreadManager`](https://discord.js.org/#/docs/main/master/class/ThreadManager?scrollTo=create), and a few more places. Although I think in some places, `create` is split in two, such as [`GuildManager`](https://discord.js.org/#/docs/main/master/class/GuildManager?scrollTo=create), [`GuildChannelManager`](https://discord.js.org/#/docs/main/master/class/GuildChannelManager?scrollTo=create), and [`GuildBanManager`](https://discord.js.org/#/docs/main/master/class/GuildBanManager?scrollTo=create).

There's also one with three arguments: [`GuildEmojiManager`](https://discord.js.org/#/docs/main/master/class/GuildEmojiManager?scrollTo=create).

Something needs to be done in a later PR to merge those in objects and increase the consistency across the library, to be honest.

iShibi replied:
I seperated `channel` because then the typedef `StageInstanceCreateOptions` can be reused in `StageChannel#createStageInstance`. The other option which Kyra suggested was to keep the `channel` but that would render it useless for the said method as it eventually gets set to `this.id` inside it. `ReplyMessageOptions` was created to address exactly this for `Message#reply`. Are we going to go back to the same thing again?

If we do want to have a single param here then I think we can create a new typedef for `options` param in `StageChannel#createStageInstance` that will not have the `channel` property.

kyranet replied:
We discussed this internally, @iCrawl says to keep them separated, and if we want to, we can merge arguments in a later PR.
